### PR TITLE
meshopt fixes and misc cleanup

### PR DESCRIPTION
**Pull request by bghgary**
The main fix is to stop using `import` in EXT_meshopt_compression which doesn't work in Babylon Native scenarios. Also added a validation test for this extension. The rest are changes to make everything more consistent.

### CODE REVIEW

**Review by sebavan — APPROVED** [review]
LGTM

File: loaders/src/glTF/2.0/Extensions/EXT_meshopt_compression.ts [hunk 1]
```diff
@@ -23,59 +30,54 @@ export class EXT_meshopt_compression implements IGLTFLoaderExtension {
...
     }
 
     /** @hidden */
     public dispose() {
+        (this._loader as any) = null;
+        delete this._decoderPromise;
     }
 
     /** @hidden */
     public loadBufferViewAsync(context: string, bufferView: IBufferView): Nullable<Promise<ArrayBufferView>> {
```
Comment (RaananW) [inline loaders/src/glTF/2.0/Extensions/EXT_meshopt_compression.ts:60]: I noticed you removed the `return null;` section (which is great!!) this means that the return signature can be changed, and that this function can be converted to an async/await function (if it makes sense).

**Review by RaananW — APPROVED** [review]
LGTM! with just a small comment

File: loaders/src/glTF/2.0/Extensions/EXT_meshopt_compression.ts [hunk 2]
```diff
@@ -23,59 +30,54 @@ export class EXT_meshopt_compression implements IGLTFLoaderExtension {
...
-                return MeshoptDecoder.ready.then(() => MeshoptDecoder);
+                return MeshoptDecoder.ready;
             });
         }
     }
 
     /** @hidden */
     public dispose() {
+        (this._loader as any) = null;
+        delete this._decoderPromise;
```
Comment (Popov72) [inline loaders/src/glTF/2.0/Extensions/EXT_meshopt_compression.ts:56]: What does calling `delete` on a promise do?

File: loaders/src/glTF/2.0/Extensions/EXT_meshopt_compression.ts (hunk 1 above)
Comment (bghgary) [reply loaders/src/glTF/2.0/Extensions/EXT_meshopt_compression.ts:60]: The helper function `LoadExtensionAsync` will still return null.

Comment (bghgary) [reply loaders/src/glTF/2.0/Extensions/EXT_meshopt_compression.ts:60]: Also, we don't want to use async/await because we target es5.

Comment (RaananW) [reply loaders/src/glTF/2.0/Extensions/EXT_meshopt_compression.ts:60]: Typescript takes care of the conversion to es5. Use async await as much as you can

Comment (sebavan) [reply loaders/src/glTF/2.0/Extensions/EXT_meshopt_compression.ts:60]: actually one of the polyfill for generators is not in for ie (I guess it is only ie) that using async await would require if I remember well the issue.

Comment (bghgary) [reply loaders/src/glTF/2.0/Extensions/EXT_meshopt_compression.ts:60]: it also generates a whole ton of code (bad for code size) and is super hard to debug in js without the map file

Comment (sebavan) [reply loaders/src/glTF/2.0/Extensions/EXT_meshopt_compression.ts:60]: that was why we were not relying on it in the code.

Comment (bghgary) [reply loaders/src/glTF/2.0/Extensions/EXT_meshopt_compression.ts:60]: I wish we can use async/await. It would simplify the code a lot.

Comment (sebavan) [reply loaders/src/glTF/2.0/Extensions/EXT_meshopt_compression.ts:60]: OHHHHH YEESSSSS ME TOOO !!!!

**Pull request merged**